client/src/pages/customer-pages/Home/index.js

```diff
@@ -55,39 +112,101 @@ class Home extends Component {
-      .then(data => {
-        sessionStorage.setItem("flight_no", data.flight_no);
-        console.log(data.flight_no);
+    const {
+      textInputValueFlightNo,
+      textInputValuepPhone,
+      checkboxValue
+    } = this.state;
 
-        if (data.isAvailable) {
-          this.setState({
-            isAvailable: true
+    if (checkboxValue === false) {
+      if (textInputValueFlightNo !== "") {
+        const data = JSON.stringify({
+          textInputValueFlightNo
+        });
+        fetch("/api/v1/check_flight", {
+          credentials: "same-origin",
+          headers: {
+            "content-type": "application/json"
+          },
+          method: "POST",
+          body: data
+        })
+          .then(response => response.json(data))
+          .then(data => {
+            sessionStorage.setItem("flight_no", data.flight_no);
+            if (data.isAvailable) {
+              this.setState({
+                isAvailable: true
+              });
+              data.isAvailable && (window.location = "/flightinfo");
+            } else {
+              this.setState({
+                isAvailable: false
+              });
+              this.invalidFlightNoAlert("Invalied Flight no.!");
+            }
+          })
+          .catch(err => {
+            console.log("There has been an error ", err);
           });
-          data.isAvailable && (window.location = "/flightinfo");
-        } else {
-          this.setState({
-            isAvailable: false
+      } else {
+        this.invalidFlightNoAlert("Invalied Flight no.!");
+      }
+    } else {
+      if (textInputValuepPhone === "") {
+        this.openModal();
+      } else {
+        const data1 = JSON.stringify({
+          textInputValuepPhone
+        });
+        fetch("/api/v1/update_customer", {
+          credentials: "same-origin",
+          headers: {
+            "content-type": "application/json"
+          },
+          method: "POST",
+          body: data1
+        })
+          .then(response => response.json(data1))
+          .then(data => {
+            if (data.isAvailable) {
+              if (textInputValueFlightNo !== "") {
+                const data = JSON.stringify({
+                  textInputValueFlightNo
+                });
+                fetch("/api/v1/check_flight", {
+                  credentials: "same-origin",
+                  headers: {
+                    "content-type": "application/json"
+                  },
+                  method: "POST",
+                  body: data
+                })
+                  .then(response => response.json(data))
+                  .then(data => {
+                    sessionStorage.setItem("flight_no", data.flight_no);
+                    if (data.isAvailable) {
+                      this.setState({
+                        isAvailable: true
+                      });
+                      data.isAvailable && (window.location = "/flightinfo");
```

Comment:
I think here (and a few other spots) you could do `this.props.history.push('/flightinfo')` to make react-router-dom take you to /flightinfo without a full page load.